### PR TITLE
[feature-porting]Onboard common and dynamic-forms modules

### PR DESCRIPTION
**Pull request by JayaShakthi97**
### Related Issues
- https://github.com/wso2/product-is/issues/16790

### Related PRs
- N/A

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: .github/workflows/pr-builder.yml [hunk 1]
```diff
@@ -196,7 +196,7 @@ jobs:
 
   build:
     name: 🚧 Build
-    needs: [ lint, typecheck ]
+    needs: [ typecheck ]
```
Comment (pavinduLakshan) [inline .github/workflows/pr-builder.yml:199]: any reason to remove lint from build prerequisites?

File: apps/console/src/features/connections/api/authenticators.ts [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+    FederatedAuthenticatorListResponseInterface, 
+    FederatedAuthenticatorMetaInterface 
+} from "../models/connection";
+
+/**
+ * Get an axios instance.
+ */
+const httpClient: HttpClientInstance = AsgardeoSPAClient.getInstance()
+    .httpRequest.bind(AsgardeoSPAClient.getInstance())
+    .bind(AsgardeoSPAClient.getInstance());
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/api/authenticators.ts:47]: Binding twice is not necessary

File: apps/console/src/features/connections/api/authenticators.ts [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+    const requestConfig: AxiosRequestConfig = {
+        headers: {
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.GET,
+        url: `${ store.getState().config.endpoints.multiFactorAuthenticators }/connectors/${ id }`
+    };
+
+    return httpClient(requestConfig)
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/api/authenticators.ts:226]: any reason not to use SWR for this new methods?

File: apps/console/src/features/connections/api/connections.ts [hunk 4]
```diff
@@ -0,0 +1,1205 @@
...
+        });
+};
+
+/**
+ * Update identity provider details.
+ *
+ * @param connection - Connection.
+ * @returns A promise containing the response.
+ */
+export const updateIdentityProviderDetails = (connection: ConnectionInterface): Promise<any> => {
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/api/connections.ts:713]: Shouldn't the method name be changed to use the term "Connection"?

**Review by pavinduLakshan — APPROVED** [review]

File: .github/workflows/pr-builder.yml (hunk 1 above)
Comment (JayaShakthi97) [reply .github/workflows/pr-builder.yml:199]: As there are a number of changed files, lint fixing is not considered under this effort. Therefore lint is removed to get the build job passed.

File: apps/console/src/features/connections/api/authenticators.ts (hunk 2 above)
Comment (JayaShakthi97) [reply apps/console/src/features/connections/api/authenticators.ts:47]: As these changes are directly brought from previous PRs, will continue as it is.

File: apps/console/src/features/connections/api/authenticators.ts (hunk 3 above)
Comment (JayaShakthi97) [reply apps/console/src/features/connections/api/authenticators.ts:226]: As these changes are directly brought from previous PRs, will continue as it is.

File: apps/console/src/features/connections/api/connections.ts (hunk 4 above)
Comment (JayaShakthi97) [reply apps/console/src/features/connections/api/connections.ts:713]: As these changes are directly brought from previous PRs, will continue as it is.

**Pull request merged**